md6/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+#![no_std]
+#![doc = include_str!("../README.md")]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/RustCrypto/media/6ee8e381/logo.svg",
+    html_favicon_url = "https://raw.githubusercontent.com/RustCrypto/media/6ee8e381/logo.svg"
+)]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![warn(missing_docs, rust_2018_idioms)]
+
+mod compress;
+pub(crate) mod consts;
+mod md6;
+
+use digest::{
+    consts::{U16, U28, U32, U48, U64, U8},
+    core_api::{CoreWrapper, CtVariableCoreWrapper, RtVariableCoreWrapper},
+};
+
+pub use digest::{Digest, Update, VariableOutput};
+
+use crate::md6::Md6VarCore;
+
+/// Md6 which allows variable output size at runtime
+pub type Md6Var = RtVariableCoreWrapper<Md6VarCore>;
+/// Core hash function for Md6 generic over output size
+pub type Md6Core<OutSize> = CtVariableCoreWrapper<Md6VarCore, OutSize>;
+/// Md6 generic over output size.
+pub type Md6<OutSize> = CoreWrapper<Md6Core<OutSize>>;
+/// Md6 with 64-bit output
+pub type Md6_64 = CoreWrapper<Md6Core<U8>>;
+/// Md6 with 128-bit output
+pub type Md6_128 = CoreWrapper<Md6Core<U16>>;
```

Comment:
I don't think we need aliases for MD6-64 and MD6-128. In the NIST submission they write the following:
>These are the MD6 versions most relevant for SHA-3: MD6-224, MD6-256, MD6-384, MD6-512.